ocean_lib/models/datatoken.py

```diff
@@ -306,6 +307,35 @@ def build_access_service(
             consumer_parameters=consumer_parameters,
         )
 
+    def dispense_if_free(self, wallet):
+        template_id = self.getId()
+
+        if template_id != 1:
+            # only free priced can use dispensers
+            return
+
+        bal = Web3.fromWei(self.balanceOf(wallet.address), "ether")
+        if bal >= 1.0:  # we're good
```

Comment:
if I already have 1.0 DT, it will skip the dispense flow.  is this what we want, cause function is named "dispense_if_free", and I assume this will be used by user to get 1 DT

Reply:
it does, because dispense_if_free returns without doing anything if the balance is already >1

Reply:
I will rename the function to dispense_if_needed